libs/libcommon/tests/test_orchestrator.py

```diff
@@ -194,6 +202,33 @@ def test_finish_job(
     assert cached_response.dataset_git_revision == REVISION_NAME
 
 
+def populate_queue() -> None:
+    Queue().create_jobs(
+        [
+            artifact_id_to_job_info(ARTIFACT_CA_1),
+            artifact_id_to_job_info(ARTIFACT_CA_2),
+            artifact_id_to_job_info(ARTIFACT_DH),
+            artifact_id_to_job_info(ARTIFACT_SA_1_1),
+            artifact_id_to_job_info(ARTIFACT_SA_1_2),
+        ]
+        * 50
+    )
+
+
+@pytest.mark.limit_memory("1.4 MB")  # Success, it uses ~1.4 MB
+def test_get_pending_jobs_df() -> None:
+    populate_queue()
+    pending_jobs_df = Queue().get_pending_jobs_df(dataset=DATASET_NAME)
+    assert pending_jobs_df.shape == (250, 9)
+
+
+@pytest.mark.limit_memory("1.6 MB")  # Will fail, it uses ~1.6 MB
```

Comment:
For `get_pending_jobs_df,` the percentage reduction is ~12%; I will leave the team to decide if this is a good indicator to switch to pymongoarrow.
The most impactful is that after getting the data frame, it is needed to perform other transformations (Like renaming _id -> job_id and making status and priority a Categorical value to be used later in a sort statement in [here](https://github.com/huggingface/dataset-viewer/blob/main/libs/libcommon/src/libcommon/state.py#L53).
There surely is another way to handle the state logic, but I would like to make a few changes initially and improve progressively once we are sure that nothing breaks with the switch to pymongoarrow.